TestProjects/VisualEffectGraph_HDRP/Assets/AllTests/Editor/Tests/VFXGUITests.cs

```diff
@@ -135,133 +97,230 @@ void CreateDataEdges(VFXContextController updateContext, List<VFXParameter> para
+            CreateAllBlocks(viewController, param.destContext, param.blocks);
         }
 
+        //Reduced selection for CreateAllDataEdgesTest, only testing block outputs
+        static CreateAllBlockParam[] kCreateAllBlockParamOutput = kCreateAllBlockParam.Where(o => o.destContext.model.contextType == VFXContextType.Output).ToArray();
         [Test]
-        public void CreateAllEventBlocksTest()
+        public void CreateAllDataEdgesTest([ValueSource(nameof(kCreateAllBlockParamOutput))] CreateAllBlockParam param)
         {
-            EditTestAsset(0);
-            CreateAllBlocks(VFXContextType.Event);
+            var viewController = StartEditTestAsset();
+            var contextController = CreateAllBlocks(viewController, param.destContext, param.blocks);
+            CreateDataEdges(viewController, contextController, CreateAllParameters(viewController));
         }
 
-        VFXContextController CreateAllBlocks(VFXContextType type)
+        VFXContextController CreateAllBlocks(VFXViewController viewController, VFXModelDescriptor<VFXContext> context, IEnumerable<VFXModelDescriptor<VFXBlock>> blocks)
         {
-            var initContextDesc = VFXLibrary.GetContexts().Where(t => t.model.contextType == type).First();
-
-            var newContext = m_ViewController.AddVFXContext(new Vector2(300, 2000), initContextDesc);
-
-            m_ViewController.ApplyChanges();
-
-            var contextController = m_ViewController.nodes.Where(t => t is VFXContextController && (t as VFXContextController).model == newContext).First() as VFXContextController;
+            var newContext = viewController.AddVFXContext(new Vector2(300, 2000), context);
+            viewController.ApplyChanges();
 
+            var contextController = viewController.nodes.Where(t => t is VFXContextController && (t as VFXContextController).model == newContext).First() as VFXContextController;
             Assert.AreEqual(contextController.model, newContext);
 
             // Adding every block compatible with an init context
-
             var newBlocks  = new List<VFXBlock>();
-            foreach (var block in VFXLibrary.GetBlocks().Where(t => t.AcceptParent(newContext)))
+            foreach (var block in blocks)
             {
                 var newBlock = block.CreateInstance();
                 contextController.AddBlock(0, newBlock);
                 newBlocks.Add(newBlock);
             }
 
-            m_ViewController.ApplyChanges();
-
+            viewController.ApplyChanges();
             foreach (var newBlock in newBlocks)
             {
                 Assert.AreEqual(contextController.blockControllers.Where(t => t.model == newBlock).Count(), 1, "Failing Block" + newBlock.name + "in context" + newContext.name);
-
                 var blockController = contextController.blockControllers.Where(t => t.model == newBlock).First() as VFXBlockController;
-
                 Assert.NotNull(blockController);
```

Comment:
I have the feeling that this check is useless because First() will throw instead of returning null

Reply:
Fix at 9aa6d748fd15fe33e5f0fad881dffe63919e234f (but removed afterwards)